components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/src/main/java/org/wso2/carbon/identity/rest/api/user/authorized/apps/v2/core/AuthorizedAppsService.java

```diff
@@ -222,6 +227,75 @@ public AuthorizedAppDTO listUserAuthorizedAppsByAppId(User user, String applicat
         return authorizedAppDTOS;
     }
 
+    /**
+     * Delete issued tokens for a given application id.
+     *
+     * @param applicationId Application Id.
+     */
+    public void deleteIssuedTokensByAppId(String applicationId) {
+
+        String tenantDomain = ContextLoader.getTenantDomainFromContext();
+        InboundAuthenticationRequestConfig reqConfig = getInboundAuthRequestConfig(applicationId, tenantDomain);
+        String clientId = reqConfig.getInboundAuthKey();
+
+        OAuthAppRevocationRequestDTO oAuthAppRevocationRequestDTO = new OAuthAppRevocationRequestDTO();
+        oAuthAppRevocationRequestDTO.setApplicationName(applicationId);
+        oAuthAppRevocationRequestDTO.setConsumerKey(clientId);
+        oAuthAppRevocationRequestDTO.setTenantDomain(tenantDomain);
+        try {
+            oAuthAdminService.revokeIssuedTokensByApplication(oAuthAppRevocationRequestDTO);
+        } catch (IdentityOAuthAdminException e) {
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_REVOKE_TOKEN_BY_APP_ID, applicationId, tenantDomain);
+        }
+    }
+
+    private InboundAuthenticationRequestConfig getInboundAuthRequestConfig(String applicationId, String tenantDomain) {
+
+        ServiceProvider application = getServiceProvider(applicationId, tenantDomain);
+        // Extract the inbound authentication request config for the given inbound type.
+        InboundAuthenticationRequestConfig inboundAuthenticationRequestConfig =
+                getInboundAuthenticationRequestConfig(application, OAUTH2);
+
+        if (inboundAuthenticationRequestConfig == null) {
+            // This means the inbound is not configured for the particular app.
+            throw handleError(Response.Status.NOT_FOUND, Constants.ErrorMessages.ERROR_CODE_INVALID_INBOUND_PROTOCOL,
+                    applicationId, tenantDomain);
+        }
+        return inboundAuthenticationRequestConfig;
+    }
+
+    private ServiceProvider getServiceProvider(String applicationId, String tenantDomain) {
+
+        try {
+            ServiceProvider application = applicationManagementService.getServiceProvider(applicationId, tenantDomain);
+            if (application == null) {
+                throw handleError(Response.Status.NOT_FOUND, Constants.ErrorMessages.ERROR_CODE_APPLICATION_NOT_FOUND,
+                        applicationId, tenantDomain);
+            }
+            return application;
+        } catch (IdentityApplicationManagementException e) {
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_GETTING_APPLICATION_INFORMATION, applicationId, tenantDomain);
+        }
+    }
+
+    private InboundAuthenticationRequestConfig getInboundAuthenticationRequestConfig(ServiceProvider application,
+                                                                                     String inboundType) {
```

Comment:
since the `inboundType` will always be "oauth2", there is no need to pass the value as a parameter here